packages/react-server/core/components/RootContainer.js

```diff
@@ -23,9 +23,11 @@ RootContainer.defaultProps = {
 }
 
 RootContainer.flattenForRender = function(element) {
-	return [{containerOpen: getRootElementAttributes(element)}]
+	let attrs = getRootElementAttributes(element) || {};
+	attrs.tagName = attrs.tagName || 'div';
+	return [{containerOpen: true, attrs: attrs}]
 		.concat(prepChildren(element))
-		.concat([{containerClose: true}])
+		.concat([{containerClose: true, attrs: attrs}])
```

Comment:
Then this could just be `{containerClose: tagName}`, and the (irrelevant for a close tag) `attrs` can be omitted.